File: AntiClown.Entertainment.Api.Dto/F1Predictions/F1RaceDto.cs
namespace AntiClown.Entertainment.Api.Dto.F1Predictions;

public class F1RaceDto
{
    public Guid Id { get; set; }
    public int Season { get; set; }
    public string Name { get; set; }
    public bool IsActive { get; set; }
    public bool IsOpened { get; set; }
    public bool IsSprint { get; set; }
    public F1RacePredictionConditionsDto Conditions { get; set; }
    public List<F1PredictionDto> Predictions { get; set; }
    public F1PredictionRaceResultDto Result { get; set; }

    public string FullName() => $"{Name}{(IsSprint ? " (спринт)" : string.Empty)}";
}

File: src/Web/AntiClown.Web.Front/src/Dto/F1Predictions/F1RaceDto.ts
import { F1PredictionDto } from "./F1PredictionDto";
import { F1PredictionRaceResultDto } from "./F1PredictionRaceResultDto";
import { F1RacePredictionConditionsDto } from "./F1RacePredictionConditionsDto";

export interface F1RaceDto {
  id: string;
  season: number;
  name: string;
  isActive: boolean;
  isOpened: boolean;
  isSprint: boolean;
  conditions?: F1RacePredictionConditionsDto;
  predictions: F1PredictionDto[];
  result: F1PredictionRaceResultDto;
}
